### ISSUE TITLE
Enhancement Request: Climate Temperature, Glass Surface Heating & Seat Heaters

### ISSUE THREAD

**Issue by coreywillwhat** [audiconnect/audi_connect_ha#303]
## Background
After extensive debugging, I've managed to customize the `start_climatisation` service call for my Audi, allowing for hard-coded control over temperature settings, glass surface heating, and individual seat heaters. This discovery enables a more personalized climate control experience directly from the code.

## Current Implementation
Currently, adjustments are made by directly modifying the service call within the `audi_services.py` file, specifically on line 547. The original line of code is:

```python
data = '{"action":{"type": "startClimatisation","settings": {"targetTemperature": 2941,"climatisationWithoutHVpower": true,"heaterSource": "electric","climaterElementSettings": {"isClimatisationAtUnlock": false, "isMirrorHeatingEnabled": true,}}}}'
```

## Proposed Enhancement
I have successfully modified the code to enable front seat heaters as well, showcasing the potential for variable control:

```python
data = '{"action":{"type": "startClimatisation", "settings": {"targetTemperature": 2941, "climatisationWithoutHVpower": true, "heaterSource": "electric", "climaterElementSettings": {"isClimatisationAtUnlock": false, "isMirrorHeatingEnabled": true, "zoneSettings": {"zoneSetting": [{"value": {"isEnabled": true, "position": "frontLeft"}}, {"value": {"isEnabled": true, "position": "frontRight"}}, {"value": {"isEnabled": false, "position": "rearLeft"}}, {"value": {"isEnabled": false, "position": "rearRight"}}]}}}}}'
```

### Details on Controllable Features

- **Temperature:** The temperature value is in deca-kelvin. The formula for converting from Fahrenheit to the target temperature is as follows:

    ```python
    selected_target_temperature_f = 70
    selected_target_temperature_c = int((selected_target_temperature_f - 32) * (5/9))
    target_temperature = selected_target_temperature_c * 10 + 2731
    ```

    This formula calculates "2941" for a 70°F target.

- **Glass Surface Heating:** Controlled by the boolean `isMirrorHeatingEnabled`.

- **Seat Heaters ("Auto Seat Comfort"):** The seat heaters are managed via boolean values under `zoneSettings` for each seat. My vehicle lacks rear seat heaters, so defaults for those do not affect functionality.

## Enhancement Request
Would it be possible to pass variables for temperature, glass surface heating, and seat comfort settings along with the VIN and action in the `start_climatisation` service call? Ideally, the service would use default values if none are provided. The variables could include:

- `temperature_f` and/or `temperature_c`
- `glass_surface_heating`
- `seat_comfort_front_left`
- `seat_comfort_front_right`
- `seat_comfort_rear_left`
- `seat_comfort_rear_right`

This enhancement would greatly increase the usability and flexibility of the Audi services integration, allowing users to control these settings directly from the Home Assistant front end.

**Comment by cdnninja** [comment]
Would this be something you could create a PR for?

**Comment by coreywillwhat** [comment]
My knowledge is pretty limited past this point. Not sure where to start to pull the values from the front end service call. I can give it a shot, but it will take me a lot of time and testing. I'll work on it, but wanted to put my findings out there if someone else can get to it first.

**Comment by coreywillwhat** [comment]
> Would this be something you could create a PR for?

I think I've done it. Won't be able to test on my vehicle until tomorrow. Will report back after testing.

**Comment by coreywillwhat** [comment]
> Would this be something you could create a PR for?

Tested, working, and submitted pull request. Apologies for the absolute nightmare of commits. I've never used the github process before. Let me know if you need me to submit a different way or change anything.

**Comment by coreywillwhat** [comment]
Will be implemented with #306
